src/Elastic.Markdown/Myst/ApplicableToParser.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Collections.Concurrent;
+using Elastic.Documentation.AppliesTo;
+
+namespace Elastic.Markdown.Myst;
+
+/// <summary>
+/// Service for parsing ApplicableTo objects from YAML with memoization for performance.
+/// </summary>
+public static class ApplicableToParser
+{
+	private static readonly ConcurrentDictionary<string, ApplicableTo?> ParsedCache = new();
```

Comment:
Conflicted about memoization here. 

Deserialization should be fast (the serialization routines get source generated at build time) and these objects are typically short lived so should be GC'd away quickly too. 

This does keep fewer instances around but I wonder how many cache hits we actually get with folks writing the yaml in different orders. 

`Applicability.TryParse()` might be worth memoizing since it has a much higher chance of hitting the cache. But even there there are so many `version lifecycle` permutations around memoizing might eat up more memory then its worth easing GC pressure.